mlx/directives/item_matrix_directive.py

```diff
@@ -131,6 +125,149 @@ def perform_replacement(self, app, collection):
         top_node += table
         self.replace_self(top_node)
 
+    @staticmethod
+    def _fill_table_body(tbody, rows, group):
+        """ Fills the table body with rows
+
+        Args:
+            tbody (nodes.tbody): Table body to extend
+            rows (Rows): Rows namedtuple object
+            group (str): Optional group option
+        """
+        if not group:
+            tbody += rows.sorted
+        elif group == 'top':
+            tbody += rows.uncovered
+            tbody += rows.covered
+        elif group == 'bottom':
+            tbody += rows.covered
+            tbody += rows.uncovered
```

Comment:
Wouldnt a return tbody here be better option? Then you don't need to pass in body, but you `+=` it on return part...

Reply:
This is a general question as I see below we actually have a couple of other functions with the same pattern.

Reply:
That would be better practice indeed. Done in  24abe8320517d0f5706ce037708e6c1f3b43b3c1.

Reply:
I believe that those other functions only modify the container variables passed as parameters. They wouldn't be able to create them from scratch. Refactoring them to return a container and merge containers afterwards is not feasible.